smartstruct/lib/src/annotations.dart

```diff
@@ -17,3 +18,14 @@ class Mapping {
   final String target;
   const Mapping({required this.source, required this.target});
 }
+
+/// CustomMapping to support explicit field mapping in case the mapped target do not match in name,type or target is created based on few fields
+///
+/// Annotate the method with [CustomMapping] and provide a valid targetField and mapping function f
+/// To achive type safety, override CustomMapping class with specific TargetField and SourceClass types
+class CustomMapping<TargetField, SourceClass> {
```

Comment:
Is it possible to extend the Mapping Annotation, instead of writing a new CustomMapping Annotation?
Basically it's an enhanced Mapping(...), because instead of mapping the target to another source property, we are mapping the target to another method.
The generator could just check if the source argument is available as a property in the Source Parameter, and if not check if there is a method and use this instead